MC/bin/o2dpg_sim_workflow.py

```diff
@@ -130,14 +132,77 @@ def getDPL_global_options(bigshm=False,nosmallrate=False):
            print('o2dpg_sim_workflow: Error! embedding background generator name not provided')
            exit(1)
 
-        INIBKG=''
-        if args.iniBkg!= '':
-           INIBKG=' --configFile ' + args.iniBkg
+        PROCESSBKG=args.procBkg
+        COLTYPEBKG=args.colBkg
+        ECMSBKG=float(args.eCM)
+        EBEAMABKG=float(args.eA)
+        EBEAMBBKG=float(args.eB)
+
+        if COLTYPEBKG == 'pp':
+           PDGABKG=2212 # proton
+           PDGBBKG=2212 # proton
+
+        if COLTYPEBKG == 'PbPb':
+           PDGABKG=1000822080 # Pb
+           PDGBBKG=1000822080 # Pb
+           if ECMSBKG < 0:    # assign 5.02 TeV to Pb-Pb
+              print('o2dpg_sim_workflow: Set BKG CM Energy to PbPb case 5.02 TeV')
+              ECMSBKG=5020.0
+           if GENBKG == 'pythia8':
+              PROCESSBKG = 'none'
+              print('o2dpg_sim_workflow: Process type not considered for Pythia8 PbPb')
+
+        if COLTYPEBKG == 'pPb':
+           PDGABKG=2212       # proton
+           PDGBBKG=1000822080 # Pb
+
+        if COLTYPEBKG == 'Pbp':
+           PDGABKG=1000822080 # Pb
+           PDGBBKG=2212       # proton
+
+        # If not set previously, set beam energy B equal to A
+        if EBEAMBBKG < 0 and ECMSBKG < 0:
+           EBEAMBBKG=EBEAMABKG
+           print('o2dpg_sim_workflow: Set beam energy same in A and B beams')
+           if COLTYPEBKG=="pPb" or COLTYPEBKG=="Pbp":
+              print('o2dpg_sim_workflow: Careful! both beam energies in bkg are the same')
+
+        if ECMSBKG > 0:
+           if COLTYPEBKG=="pPb" or COLTYPEBKG=="Pbp":
+              print('o2dpg_sim_workflow: Careful! bkg ECM set for pPb/Pbp collisions!')
+
+        if ECMSBKG < 0 and EBEAMABKG < 0 and EBEAMBBKG < 0:
+           print('o2dpg_sim_workflow: Error! bkg ECM or Beam Energy not set!!!')
+           exit(1)
 
         CONFKEYBKG=''
         if args.confKeyBkg!= '':
            CONFKEYBKG=' --configKeyValues ' + args.CONFKEYBKG
 
+        # Background PYTHIA configuration
+        BKG_CONFIG_task=createTask(name='genbkgconf')
```

Comment:
You probably need to add this task to the list of requirements for `BKGtask` further below.

Reply:
do you mean this? workflow['stages'].append(BKG_CONFIG_task) it is below

Reply:
No. I believe that BKG_CONFIG_task needs to finish before BKGtask can start, right? So you need to add the `needs=[]` field for BKGtask.

Reply:
I tried         BKGtask=createTask(name='bkgsim', lab=["GEANT"], needs=BKG_CONFIG_task['name'], cpu=NWORKERS )
but it does not work

Reply:
You need a `list`: `needs=[BKG_CONFIG_task['name']]`

Reply:
I think this is what I did on the last commit 
        BKGtask=createTask(name='bkgsim', lab=["GEANT"], needs=[BKG_CONFIG_task['name']], cpu=NWORKERS )

Reply:
Ok. This was not apparent from these comments. Did you test the whole workflow? Does it run as expected? If yes, I think we can go ahead.

Reply:
yes, it seems to work, although I did not let it end, too hot my computer